File: backend/src/main/java/com/yunbei/skill/controller/AuthController.java
package com.yunbei.skill.controller;

import com.yunbei.skill.common.Result;
import com.yunbei.skill.dto.ChangePasswordDTO;
import com.yunbei.skill.dto.LoginDTO;
import com.yunbei.skill.dto.UserPermissionDTO;
import com.yunbei.skill.service.AuthService;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.security.core.Authentication;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/api/auth")
@RequiredArgsConstructor
public class AuthController {

    private final AuthService authService;

    @PostMapping("/login")
    public Result<UserPermissionDTO> login(@Valid @RequestBody LoginDTO dto) {
        UserPermissionDTO user = authService.login(dto.getUsername(), dto.getPassword());
        return Result.success("登录成功", user);
    }

    @PostMapping("/logout")
    public Result<Void> logout(Authentication authentication) {
        Long userId = (Long) authentication.getPrincipal();
        authService.logout(userId);
        return Result.success();
    }

    @PostMapping("/change-password")
    public Result<Void> changePassword(Authentication authentication, @Valid @RequestBody ChangePasswordDTO dto) {
        Long userId = (Long) authentication.getPrincipal();
        authService.changePassword(userId, dto.getOldPassword(), dto.getNewPassword());
        return Result.success();
    }

    @GetMapping("/permissions")
    public Result<List<String>> getUserPermissions(Authentication authentication) {
        Long userId = (Long) authentication.getPrincipal();
        return Result.success(authService.getUserPermissions(userId));
    }
}

File: pcodex-openai/pcodex-openai/apps/api/src/modules/auth/auth.controller.ts
import {
  Body,
  Controller,
  Get,
  Patch,
  Post,
  Req,
  UseGuards,
} from "@nestjs/common";
import { CurrentUser } from "../../common/decorators/current-user.decorator";
import { JwtAuthGuard } from "../../common/guards/jwt-auth.guard";
import { AuthService } from "./auth.service";
import { ChangePasswordDto } from "./dto/change-password.dto";
import { LoginDto } from "./dto/login.dto";

@Controller("auth")
export class AuthController {
  constructor(private readonly authService: AuthService) {}

  @Post("login")
  async login(@Body() dto: LoginDto, @Req() req: { ip?: string }) {
    return {
      success: true,
      message: "登录成功",
      data: await this.authService.login(dto, req.ip),
    };
  }

  @UseGuards(JwtAuthGuard)
  @Get("me")
  async me(
    @CurrentUser()
    user: { sub: string; account: string; name: string; department: string; permissions: string[] },
  ) {
    return {
      success: true,
      message: "获取成功",
      data: await this.authService.me(user),
    };
  }

  @UseGuards(JwtAuthGuard)
  @Patch("password")
  async changePassword(
    @CurrentUser() user: { sub: string },
    @Body() dto: ChangePasswordDto,
  ) {
    return {
      success: true,
      message: "修改成功",
      data: await this.authService.changePassword(user.sub, dto),
    };
  }
}
